src/main/java/hwr/oop/riddler/model/component/CellGroup.java

```diff
@@ -1,4 +1,17 @@
 package hwr.oop.riddler.model.component;
 
-public interface CellGroup {
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public abstract class CellGroup {
+    protected Set<Cell> cells = new HashSet<>();
+
+    protected CellGroup(Cell[] cells) {
+        this.cells.addAll(List.of(cells));
+    }
+
+    public void whittleDownPossibles(){
```

Comment:
Das würde ich garnicht umbedingt in die CellGroup Logik packen, glaube ich. Wie man schlussendlich Möglichkeiten streichen kann ist ja vielfältig und es gibt verschiedene Wege.

Ich hätte folgende Idee:

Es gibt ein interface IterativeSudokuSolver, das fogende Methode enthält:
/*
Returns true if progress was made during solving the sudoku
*/
boolean solve(Sudoku sudoku);

Eine Implementierung der Klasse könnte dann z.B. FillCellBasedOnPossibleSolver heißen, die z.B. alle Cells eines Sudokus durchgeht, und einen festen Wert einträgt, sobald es nur noch einen possible gibt.

Eine weitere Implementierung des Solvers könnte z.B. durch alle CellGroups durchgehen, und basierenden auf festen Werten diesen Wert aus allen Possibles aus den anderen Cells streicht. So hätte man dann schonmal basically eine Oma die ein Kreuzworträtsel aus einer Zeitung löst implementiert. :D

Diese einzelnen Implementierung könnte man mega gut auf Funktionen testen, in dem die Anforderungen an die jeweiligen IterativeSudokuSolver sehr überschaubar gehalten werden können.

Am Ende hat man dann einen SudokSolver, der verschiedene IterativeSudokuSolver als Argument übergeben bekommen kann. Der führt dann alle IterativeSudokuSolver so lange aus, bis keiner mehr true zurück gebt, also es sich nichts mehr ändert.

Falls dir der Vorschlag gefällt, let me know ob du Lust hättest das umzusetzen oder ich eine MR dazu machen soll :)

Reply:
Jap, die Methode gehört nicht in ein model, da hast du recht. 

An den solver kannst du dich gerne schonmal setzen, ich will dir nicht alles vom Projekt wegnehmen😆. Während du das machst, kann ich mich ja schonmal an Tests setzten.